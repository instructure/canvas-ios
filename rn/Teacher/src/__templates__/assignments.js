//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const assignment: Template<Assignment> = template({
  id: '123456789',
  course_id: '987654321',
  name: 'learn to write code',
  description: 'a course that can take any beginner to professional in a matter of microseconds.  Harness the power of your favorite languages like PHP and Cold Fusion.',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
  created_at: '2017-03-17T19:15:25Z',
  updated_at: '2017-03-17T19:15:25Z',
  points_possible: 20,
  grading_type: 'points',
  assignment_group_id: '1376',
  peer_reviews: false,
  automatic_peer_reviews: false,
  position: 1,
  submission_types:
  [ 'online_text_entry',
    'online_url',
    'media_recording',
    'online_upload' ],
  needs_grading_count: 0,
  html_url: 'http://hostname.tld/whatever',
  overrides: [],
  grade_group_students_individually: false,
  group_category_id: null,
  has_overrides: false,
  only_visible_to_overrides: false,
  published: true,
  unpublishable: false,
  rubric_assessment: {},
  rubric_settings: null,
  rubric: null,
  anonymize_students: false,
})

export const assignmentGroup: Template<AssignmentGroup> = template({
  id: '1',
  name: 'Learn React Native',
  position: 1,
  assignments: [assignment({ assignment_group_id: '1' })],
  group_weight: 1,
  integration_data: {},
  sis_source_id: '',
})

export const assignmentDueDate: Template<AssignmentDate> = template({
  title: 'due',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})

export const assignmentOverride: Template<AssignmentOverride> = template({
  id: '1',
  assignment_id: '1',
  student_ids: [],
  group_id: '1',
  course_section_id: '1',
  title: 'override',
  all_day: false,
  all_date_date: '',
  due_at: '2037-06-01T05:59:00Z',
  unlock_at: '2017-03-17T06:00:00Z',
  lock_at: '2037-06-01T05:59:59Z',
})
