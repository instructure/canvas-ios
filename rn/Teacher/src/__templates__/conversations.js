//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import template, { type Template } from '../utils/template'

export const conversation: Template<Conversation> = template({
  id: '123456789',
  subject: 'learn to write code',
  workflow_state: 'read',
  last_message: 'this is the worst class ever',
  last_message_at: '2037-06-01T05:59:00Z',
  last_authored_message: 'this is the worst class ever',
  last_authored_message_at: '2037-06-01T05:59:00Z',
  participants: [
    {
      id: '1234',
      name: 'artorias of the abyss',
    },
    {
      id: '6789',
      name: 'solaire of astora',
    },
  ],
  avatar_url: 'http://www.fillmurray.com/200/300',
  context_name: 'Test Class',
  context_code: 'course_1',
  audience_contexts: {},
  audience: [],
  message_count: 2,
  private: false,
  properties: null,
  starred: false,
  subscribed: false,
  visible: true,
})

export const conversationMessage: Template<ConversationMessage> = template({
  id: '123456789',
  body: 'this is neat',
  author_id: '1234',
  created_at: '2037-06-01T05:59:00Z',
  attachments: [
    {
      id: '1',
      display_name: 'something.pdf',
      mime_class: 'pdf',
    },
  ],
  generated: false,
})
