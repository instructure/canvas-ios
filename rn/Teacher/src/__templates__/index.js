//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export * from '../redux/__templates__/app-state'
export * from './account-notification'
export * from './addressBook'
export * from './api'
export * from './assignments'
export * from './attachment'
export * from './conversations'
export * from './course'
export * from './discussion'
export * from './enrollments'
export * from './error'
export * from './external-tool'
export * from './file'
export * from './grading-periods'
export * from './group'
export * from './helm'
export * from './launch-definitions'
export * from './mediaComment'
export * from './quiz'
export * from './quizSubmission'
export * from './role'
export * from './rubric'
export * from './section'
export * from './session'
export * from './submissions'
export * from './tab'
export * from './users'
export * from './help-links'
export * from './dashboard'
export * from '../modules/assignee-picker/__template__/Assignee'
