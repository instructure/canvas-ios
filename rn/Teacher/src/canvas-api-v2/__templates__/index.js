// @flow
export * from './submission'
export * from './user'
export * from './assignment'
export * from './course'
export * from './group'
export * from './section'
export * from './groupSet'

// QUERIES
export * from '../queries/__templates__/SubmissionList'
export * from '../queries/__templates__/ContextCard'
