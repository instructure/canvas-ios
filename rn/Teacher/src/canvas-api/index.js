//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

export { default } from './apis/index'
export * from './apis/assignmentGroups'
export * from './apis/assignments'
export * from './apis/conversations'
export * from './apis/courses'
export * from './apis/enrollments'
export * from './apis/external-tools'
export * from './apis/groups'
export * from './apis/login'
export * from './apis/quizzes'
export * from './apis/submissions'
export * from './apis/users'
export * from './apis/media-comments'
export * from './apis/feature-flags'
export * from './apis/file-uploads'
export * from './apis/accounts'
export * from './apis/dashboard'

export * from './session'

export { default as httpClient, isAbort, httpCache } from './httpClient'
