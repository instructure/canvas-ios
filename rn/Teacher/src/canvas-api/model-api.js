//
// This file is part of Canvas.
// Copyright (C) 2018-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow
export * from './fetch-props-for'
export * from './model'
export { httpCache } from './httpClient'

import httpClient, { httpCache } from './httpClient'
import parseLink from './utils/parse-link-header'
import {
  CourseModel,
  PageModel,
  ToDoModel,
} from './model'

type ApiPolicy = 'cache-only' | 'cache-and-network' | 'network-only'
type ApiOptions = {|
  policy: ApiPolicy,
  onStart?: ApiPromise<any> => void,
  onComplete?: ApiPromise<any> => void,
  onAnyComplete?: ApiPromise<any> => void,
  onError?: Error => void,
|}
type Paginated<T> = {|
  list: T,
  next: ?string,
  getNextPage: ?(() => ApiPromise<Paginated<T>>),
|}

const emptyArray = []
const emptyPaginated: Paginated<any> = {
  list: emptyArray,
  next: null,
  getNextPage: null,
}

export class API {
  options: ApiOptions

  constructor (options: ApiOptions) {
    this.options = options
  }

  cleanup () {
    this.options.onStart =
    this.options.onComplete =
    this.options.onAnyComplete =
    this.options.onError = undefined
  }

  get (url: string, config: ApiConfig = {}) {
    const { policy } = this.options
    const cached = httpCache.get(url, config)
    if (
      policy === 'network-only' ||
      (policy === 'cache-and-network' && cached.expiresAt < Date.now())
    ) {
      this.request(httpClient.get(url, config))
    }
    return cached.value || null
  }

  paginate<T> (url: string, config: ApiConfig = {}): Paginated<T> {
    const transform = config.transform
    config.transform = (value: any, response: ApiResponse<any>) => {
      const list = transform ? transform(value, response) : value
      const { current, first, next } = parseLink(response.headers.link) || {}
      const getNextPage = next && (() => {
        this.request(httpClient.get(next, config))
      })
      if (getNextPage && config.params && config.params.per_page >= 100) {
        getNextPage() // automatically request all pages
      }
      if (current && current !== first) {
        const base = httpCache.get(url, config).value
        if (base) {
          base.list = base.list.concat(list)
          base.next = next
          base.getNextPage = getNextPage
        }
        return null
      }
      return {
        list,
        next,
        getNextPage,
      }
    }

    const cached = httpCache.get(url, config)
    const { policy } = this.options
    if (
      policy === 'network-only' ||
      (policy === 'cache-and-network' && cached.expiresAt < Date.now())
    ) {
      this.request(httpClient.get(url, config))
    }
    // getNextPage won't survive serialization
    const value = cached.value
    if (value && value.next && !value.getNextPage && policy !== 'cache-only') {
      value.getNextPage = () => this.request(httpClient.get(value.next, config))
    }
    return value || emptyPaginated
  }

  post (url: string, data: *, config: ApiConfig = {}) {
    return this.request(httpClient.post(url, data, config))
  }

  put (url: string, data: *, config: ApiConfig = {}) {
    return this.request(httpClient.put(url, data, config))
  }

  delete (url: string, config: ApiConfig = {}) {
    return this.request(httpClient.delete(url, config))
  }

  request (promise: ApiPromise<any>) {
    const { onStart, onComplete, onError } = this.options
    onStart && onStart(promise)
    promise
      .catch(error => { onError && onError(error) })
      .then(() => { onComplete && onComplete(promise) })
    return promise
  }

  getCourseColor (courseID: string): string {
    const json: ?Object = this.get('users/self/colors')
    return json && json.custom_colors[`course_${courseID}`] || '#aaa'
  }

  getCourse (courseID: string): ?CourseModel {
    return this.get(`courses/${courseID}`, {
      params: {
        include: [ 'permissions', 'term', 'favorites', 'course_image', 'sections' ],
      },
      transform: (course: Course) => new CourseModel(course),
    })
  }

  getPages (context: CanvasContext, contextID: string): Paginated<PageModel[]> {
    return this.paginate(`${context}/${contextID}/pages`, {
      params: {
        per_page: 100,
      },
      transform: (pages: Page[]) => pages.map(page => new PageModel(page)),
    })
  }

  createPage (context: CanvasContext, contextID: string, parameters: PageParameters): ApiPromise<PageModel> {
    return this.post(`${context}/${contextID}/pages`, { wiki_page: parameters }, {
      transform: (page: Page) => new PageModel(page),
    })
  }

  getPage (context: CanvasContext, contextID: string, url: string): ?PageModel {
    return this.get(`${context}/${contextID}/pages/${url}`, {
      transform: (page: Page) => new PageModel(page),
    })
  }

  getFrontPage (contextID: string): ?PageModel {
    return this.get(`courses/${contextID}/front_page`, {
      transform: (page: Page) => new PageModel(page),
    })
  }

  updatePage (context: CanvasContext, contextID: string, url: string, parameters: PageParameters): ApiPromise<PageModel> {
    return this.put(`${context}/${contextID}/pages/${url}`, { wiki_page: parameters }, {
      transform: (page: Page) => new PageModel(page),
    })
  }

  deletePage (context: CanvasContext, contextID: string, url: string): ApiPromise<null> {
    return this.delete(`${context}/${contextID}/pages/${url}`)
  }

  getToDos (): Paginated<ToDoModel[]> {
    return this.paginate('users/self/todo', {
      transform: (todos: ToDoItem[]) => todos.map(todo => new ToDoModel(todo)),
    })
  }
}
