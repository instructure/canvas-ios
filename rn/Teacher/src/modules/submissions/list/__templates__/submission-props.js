//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import template, { type Template } from '../../../../utils/template'
import type { SubmissionDataProps } from '../submission-prop-types'

const templates = {
  ...require('../../../../__templates__/submissions'),
}

const submissionWithHistory = templates.submissionHistory()

// $FlowFixMe
export const submissionProps: Template<SubmissionDataProps> = template({
  grade: 'B-',
  name: submissionWithHistory.user.name,
  status: 'submitted',
  userID: submissionWithHistory.user.id,
  avatarURL: submissionWithHistory.user.avatar_url,
  sectionID: '1',
})
