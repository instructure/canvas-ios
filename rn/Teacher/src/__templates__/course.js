//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

/* @flow */

import { CourseModel } from '../canvas-api/model'
import template, { type Template } from '../utils/template'
import { section } from './section'

export const courseTerm: Template<CourseTerm> = template({
  id: '1',
  created_at: '2017-03-17T19:15:25Z',
  name: 'Default Term',
})

export const course: Template<Course> = template({
  id: '1',
  account_id: '1',
  name: 'Learn React Native',
  short_name: 'rn',
  course_code: 'rn 101',
  image_download_url: 'https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg',
  is_favorite: true,
  default_view: 'wiki',
  enrollments: [{
    enrollment_state: 'active',
    role: 'TeacherEnrollment',
    role_id: '1',
    type: 'teacher',
    user_id: '1',
  }],
  permissions: {
    viewAnalytics: true,
    viewAllGrades: true,
  },
  workflow_state: 'available',
  term: {
    id: '1',
    created_at: '2018-02-08T23:13:39Z',
    end_at: '2030-02-08T23:13:39Z',
    start_at: '2018-02-08T23:13:39Z',
    name: 'Default Term',
    workflow_state: 'available',
  },
  end_at: '2030-02-08T23:13:39Z',
})

export const courseModel: Template<CourseModel> = overrides =>
  Object.assign(new CourseModel(course()), overrides)

export const courseWithSection: Template<Course> = function (defaults) {
  return course({
    ...defaults,
    sections: [section()],
  })
}

export const customColors: Template<CustomColors> = template({
  custom_colors: {
    course_1: '#fff',
  },
})

export const courseSettings: Template<CourseSettings> = template({
  allow_student_forum_attachments: true,
})
