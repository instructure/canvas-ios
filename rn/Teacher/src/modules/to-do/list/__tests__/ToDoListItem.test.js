//
// This file is part of Canvas.
// Copyright (C) 2017-present  Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//

// @flow

import { shallow } from 'enzyme'
import React from 'react'
import { API, httpCache } from '../../../../canvas-api/model-api'
import * as template from '../../../../__templates__'
import Connected, { ToDoListItem } from '../ToDoListItem'

describe('ToDoListItem', () => {
  let props
  beforeEach(() => {
    httpCache.clear()
    props = {
      navigator: template.navigator(),
      item: template.toDoModel({ courseID: '1' }),
      courseName: 'Course 1',
      courseColor: '#fff',
      api: new API({ policy: 'cache-only' }),
      isLoading: false,
      isSaving: false,
      loadError: null,
      saveError: null,
      refresh: jest.fn(),
      onPress: jest.fn(),
    }
  })

  it('gets courseColor and courseName from the model api', () => {
    const courseColor = 'green'
    const course = template.courseModel()
    httpCache.handle('GET', 'users/self/colors', { custom_colors: { course_1: courseColor } })
    httpCache.handle('GET', 'courses/1', course)
    const tree = shallow(<Connected item={props.item} onPress={jest.fn()} />)
    expect(tree.find(ToDoListItem).props()).toMatchObject({
      courseColor,
      courseName: course.name,
    })
  })

  it('renders', () => {
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders published assignment', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', published: true }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders unpublished assignment', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', published: false }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders quiz', () => {
    props.item = template.toDoModel({
      assignment: null,
      quiz: template.quiz({ title: 'Quiz 1' }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders discussion', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ name: 'Assignment 1', submission_types: ['discussion_topic'] }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('renders null due date', () => {
    props.item = template.toDoModel({
      assignment: template.assignment({ due_at: null }),
    })
    const tree = shallow(<ToDoListItem {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('passes the item to onPress', () => {
    const tree = shallow(<ToDoListItem {...props} />)
    tree.find('Row').simulate('Press')
    expect(props.onPress).toHaveBeenCalledWith(props.item)
  })
})
