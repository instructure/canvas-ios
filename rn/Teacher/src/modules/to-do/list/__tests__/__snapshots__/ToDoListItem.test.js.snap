// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoListItem renders 1`] = `
<Row
  accessibilityLabel="Published, learn to write code, Course 1, Due May 31, 2037 at 11:59 PM, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="learn to write code"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      Due May 31, 2037 at 11:59 PM
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;

exports[`ToDoListItem renders discussion 1`] = `
<Row
  accessibilityLabel="Published, Assignment 1, Course 1, Due May 31, 2037 at 11:59 PM, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="Assignment 1"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      Due May 31, 2037 at 11:59 PM
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;

exports[`ToDoListItem renders null due date 1`] = `
<Row
  accessibilityLabel="Published, learn to write code, Course 1, No Due Date, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="learn to write code"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      No Due Date
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;

exports[`ToDoListItem renders published assignment 1`] = `
<Row
  accessibilityLabel="Published, Assignment 1, Course 1, Due May 31, 2037 at 11:59 PM, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="Assignment 1"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      Due May 31, 2037 at 11:59 PM
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;

exports[`ToDoListItem renders quiz 1`] = `
<Row
  accessibilityLabel="Published, Quiz 1, Course 1, Due Jan 23, 2013 at 11:59 PM, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="Quiz 1"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      Due Jan 23, 2013 at 11:59 PM
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;

exports[`ToDoListItem renders unpublished assignment 1`] = `
<Row
  accessibilityLabel="Not Published, Assignment 1, Course 1, Due May 31, 2037 at 11:59 PM, 5 NEED GRADING"
  disclosureIndicator={true}
  onPress={[Function]}
  renderImage={[Function]}
  testID="to-do.list.https://api.instructure.com/api/v1/path/to/html_url.row"
  title="Assignment 1"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Course 1
    </Text>
    <Text
      style={
        Object {
          "color": "#8B969E",
          "fontSize": 13,
          "marginTop": 3,
        }
      }
    >
      Due May 31, 2037 at 11:59 PM
    </Text>
    <Text
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 9,
            "borderWidth": 1,
            "flex": 0,
            "fontSize": 11,
            "fontWeight": "600",
            "marginTop": 4,
            "overflow": "hidden",
            "paddingBottom": 1,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "borderColor": "#008EE2",
            "color": "#008EE2",
          },
        ]
      }
    >
      5 NEED GRADING
    </Text>
  </View>
</Row>
`;
